protocol/x/vault/keeper/shares.go

```diff
@@ -47,3 +56,72 @@ func (k Keeper) getTotalSharesIterator(ctx sdk.Context) storetypes.Iterator {
 
 	return storetypes.KVStorePrefixIterator(store, []byte{})
 }
+
+// MintShares mints shares of a vault for `owner` based on `quantumsToDeposit` by:
+// 1. Increasing total shares of the vault.
+// 2. Increasing owner shares of the vault for given `owner`.
+func (k Keeper) MintShares(
+	ctx sdk.Context,
+	vaultId types.VaultId,
+	owner string,
+	quantumsToDeposit *big.Int,
+) error {
+	// Quantums to deposit should be positive.
+	if quantumsToDeposit.Cmp(big.NewInt(0)) <= 0 {
+		return types.ErrInvalidDepositAmount
+	}
+	totalShares, exists := k.GetTotalShares(ctx, vaultId)
+	existingTotalShares := totalShares.NumShares.BigInt()
+	var sharesToMint *big.Int
+	if !exists || existingTotalShares.Cmp(big.NewInt(0)) < 1 {
+		// Mint `quoteQuantums` number of shares.
+		sharesToMint = quantumsToDeposit
+		// Initialize existingTotalShares as 0.
+		existingTotalShares = big.NewInt(0)
+	} else {
+		// Get vault equity.
+		equity, err := k.GetVaultEquity(ctx, vaultId)
+		if err != nil {
+			return err
+		}
+		// Don't mint shares if equity is non-positive.
+		if equity.Cmp(big.NewInt(0)) <= 0 {
+			return types.ErrNonPositiveEquity
+		}
+		// Mint `deposit (in quote quantums) * existing shares / vault equity (in quote quantums)`
+		// number of shares.
+		// For example:
+		// - a vault currently has 5000 shares and 4000 equity (in quote quantums)
+		// - each quote quantum is worth 5000 / 4000 = 1.25 shares
+		// - a deposit of 1000 quote quantums should thus be given 1000 * 1.25 = 1250 shares
```

Comment:
This is no longer accurate right? Due to the scaling logic the # of shares would be scaled by 4 (5000/4000 -> 5/4). So there would be 20,000 shares now in the vault and 5000 shares would be minted?